common/src/main/java/org/dash/wallet/common/util/GenericUtils.java

```diff
@@ -179,4 +183,19 @@ public static String formatFiatWithoutComma(String fiatValue){
         boolean fiatValueContainsCommaWithDecimal = fiatValue.contains(",") && fiatValue.contains(".");
         return fiatValueContainsCommaWithDecimal ? fiatValue.replaceAll(",", "") :  fiatValue.replaceAll(",", ".");
     }
+
+    public static String fiatToString(Fiat fiat) {
+        MonetaryFormat format = Constants.SEND_PAYMENT_LOCAL_FORMAT.noCode();
+        NumberFormat numberFormat = NumberFormat.getCurrencyInstance(getDeviceLocale());
+        Currency currency = Currency.getInstance(fiat.currencyCode);
+        numberFormat.setCurrency(currency);
+        String currencySymbol = currency.getSymbol(getDeviceLocale());
+        boolean isCurrencyFirst = numberFormat.format(1.0).startsWith(currencySymbol);
+
+        if (isCurrencyFirst) {
+            return currencySymbol + " " + format.format(fiat);
+        } else {
+            return format.format(fiat) + " " + currencySymbol;
+        }
```

Comment:
I wonder if this works correctly with RTL languages.

Reply:
I'll check

Reply:
Well, the result matches what android APIs are producing, f.e. "100 ₫" which then gets reverted to "₫ 100" when assigning this to a `TextView`, which I assume is what needs to be happening.
However, we do have some minor problems with RTL all around the app. We probably should have a story to investigate and fix those.

Reply:
That is fine.